src/modules-load/modules-load.c

```diff
@@ -52,12 +53,30 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
         return 0;
 }
 
-static int apply_file(struct kmod_ctx *ctx, const char *path, bool ignore_enoent) {
+static void *worker_thread(void *arg) {
+        int sock_fd = *(int *) arg;
+        char buffer[32];
+        ssize_t bytes_received;
+        _cleanup_(sym_kmod_unrefp) struct kmod_ctx *ctx = NULL;
+        int r = module_setup_context(&ctx);
+        if (r < 0)
+                log_error_errno(r, "Failed to initialize libkmod context: %m");
+
+        while ((bytes_received = recv(sock_fd, buffer, sizeof(buffer), 0)) > 0) {
```

Comment:
please add some error handling here regarding datagram size, i.e. maybe set MSG_TRUNC, and then write a quick log message if the returned size s larger than the buffer we pass, and continue.